### ISSUE TITLE
Override data after deleting a translation

### ISSUE THREAD

**Issue by FarhangQ** [tractorcow-farm/silverstripe-fluent#847]
### Module version(s) affected

6.0

### Description

test case:
create a job in DE
then copy it to EN and change text in english to english
then use translations button to publish
until here, everything works
then use unpublish all languages from translations button.
then check language Versions, both are in english.
![Bildschirmfoto 2024-05-03 um 15 10 10](https://github.com/tractorcow-farm/silverstripe-fluent/assets/11559624/829223a2-55eb-48a2-8fe9-9798cdc7e4a7)

it saved the english version to the german one

### How to reproduce

-

### Validations

- [X] Check that there isn't already an issue that reports the same bug
- [X] Double check that your reproduction steps work in a fresh installation of [`silverstripe/installer`](https://github.com/silverstripe/silverstripe-installer) (with any code examples you've provided)

**Comment by tractorcow** [comment]
Could it be that "unpublish all languages" is deleting not only the live localisation, but also the stage at the same time by accident?

**Comment by FarhangQ** [comment]
> Could it be that "unpublish all languages" is deleting not only the live localisation, but also the stage at the same time by accident?

No, It just delete the language that we deleted then after trying to add new translation of it ( the non-deleted one ), the new translation will overwritten the main source.
For ex:
1. English
2. German
3. Removing German
4. Adding German
5. Now it'll change English exactly like German content
